### PR TITLE
feat(`@vtmn/css`, `@vtmn/svelte`, `@vtmn/react`, `@vtmn/vue`): rework `VtmnNavbarLink` for SSR

### PR DESCRIPTION
**Pull request by Tlahey**
BREAKING CHANGE: 'VtmnNavbarLink' remove properties label and showLabel, replaced by a span slot

## Changes description
Move the Label property into slot in order to apply a `sr-only` with breakpoints.

## Context
With SSR, it is not possible to detect the width of the device. So, in order to avoid flikering effects, we have to set the label on a slot in order to apply style show/hidden on it.

## How it looks
image
image
image
image

On the navbar
image
image

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [ ] If it includes design changes, please ask for a review `design-system-core-team-design` GitHub team.

## Does this introduce a breaking change?

- Yes

## Other information
Rework the css part and enhancer the CSS in order to match the figma with the VtmnBadge.

Current : 
image

Target : 
image
image

### CODE REVIEW

File: packages/showcases/svelte/stories/components/navigation/VtmnNavbarLink/VtmnNavbarLink.stories.svelte [hunk 1]
```diff
@@ -19,19 +19,48 @@
...
+    width: 1px;
+    height: 1px;
+    padding: 0;
+    margin: -1px;
+    overflow: hidden;
+    clip: rect(0, 0, 0, 0);
+    white-space: nowrap;
+    border-width: 0;
+  }
+</style>
```
Comment (Tlahey) [inline packages/showcases/svelte/stories/components/navigation/VtmnNavbarLink/VtmnNavbarLink.stories.svelte:66]: Can be remove

**Review by lauthieb — APPROVED** [review]
Thanks @Tlahey, for me that's ok but still a difference for the badge position: 
CleanShot 2023-04-27 at 15 50 56@2x
http://localhost:6006/?path=/story/components-navigation-navbar--overview

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @Tlahey, for me that's ok but still a difference for the badge position: 
CleanShot 2023-04-27 at 15 50 56@2x
http://localhost:6006/?path=/story/components-navigation-navbar--overview

**Review by lauthieb — DISMISSED** [review]
That's better! Thanks @Tlahey

File: packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx [hunk 2]
```diff
@@ -34,35 +24,24 @@ export interface VtmnNavbarLinkProps
...
-      className={clsx(
-        'vtmn-navbar_link',
-        'vtmn-navbar_link--icon-alone',
-        className,
-      )}
-      {...props}
-    >
-      {children}
+      {childrenArray.find((child: ReactElement) => {
+        console.log(child);
```
Comment (thibault-mahe) [inline packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:40]: to be deleted :p

File: packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx [hunk 3]
```diff
@@ -34,35 +24,24 @@ export interface VtmnNavbarLinkProps
...
+        return child.type === VtmnBadge;
+      })}
       <VtmnIcon value={icon} aria-hidden={true} />
-      <span className="vtmn-sr-only">{label}</span>
+      {childrenArray.find((child: ReactElement) => child.type === 'span')}
```
Comment (thibault-mahe) [inline packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:44]: Better be sure that this condition can always be fulfilled... Should we add a fallback here ?

**Review by thibault-mahe — APPROVED** [review]

File: packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx (hunk 2 above)
Comment (lauthieb) [reply packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:40]: ping @Tlahey :)

File: packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx (hunk 3 above)
Comment (lauthieb) [reply packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:44]: ping @Tlahey :)

Comment (Tlahey) [reply packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:44]: On other framework no fallback options are defined.
We should explained on the doc that the span are required with label.

Comment (Tlahey) [reply packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:43]: I check if undefined return an error, if not, I return an empty string

Comment (Tlahey) [reply packages/sources/react/src/components/navigation/VtmnNavbarLink/VtmnNavbarLink.tsx:43]: image

With react, without span it's good

**Review by lauthieb — DISMISSED** [review]

**Review by thibault-mahe — APPROVED** [review]

**Comment by C0ZEN** [comment]
Since it's a breaking change, how come it was released as `2.3.0` instead of `3.0.0`?

**Comment by Tlahey** [comment]
Totally agree, with the `BREAKING CHANGE` label, should have updated as a major version 🤔

**Comment by thibault-mahe** [comment]
I agree with you, something went wrong. I'm reverting the commits until I find some times to check on this: https://github.com/Decathlon/vitamin-web/pull/1435 

Thanks @C0ZEN  :)

**Pull request merged**